### PR TITLE
BE: reset dynamo before each test in test_module.py

### PR DESCRIPTION
**Pull request by shunting314**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131397
* __->__ #131372
* #131388
* #131551

https://github.com/pytorch/pytorch/pull/126586 tried to reset dynamo before each unit test. That PR get reverted a couple of times because we see post-land test failures that we don't see before merge. This PR only reset dynamo before each tests in `test_module.py` to make it easier to land.

Eventually after we reset dynamo in each individual test files, we can move the change to the base class (TestCase) and remove the change in individual test files.

### CODE REVIEW

File: test/test_modules.py [hunk 1]
```diff
@@ -25,6 +25,13 @@ class TestModule(TestCase):
     precision = 1e-5
     rel_tol = 1e-5
 
+    def setUp(self):
+        TestCase.setUp(self)
```
Comment (zou3519) [inline test/test_modules.py:29]: nit: super().setUp ?

**Review by zou3519 — APPROVED** [review]

File: test/test_modules.py (hunk 1 above)
Comment (shunting314) [reply test/test_modules.py:29]: That's the first thing I tried :) But it fail with error:
```
TypeError: super(type, obj): obj must be an instance or subtype of type
```
I think it's due to the fact that we call instantiate_device_type_tests to create device special class'es

Comment (zou3519) [reply test/test_modules.py:29]: Fair enough

**Review by zou3519 — CHANGES_REQUESTED** [review]
Sorry, I changed my assessment: see the comments in https://github.com/pytorch/pytorch/pull/131388

**Review by zou3519 — APPROVED** [review]

**Comment by kit1980** [comment]
This broke https://github.com/pytorch/pytorch/actions/runs/10149118852/job/28065175173

**Comment by zou3519** [comment]
@kit1980 do you know why there was no signal on this PR?

**Comment by zou3519** [comment]
> @kit1980 do you know why there was no signal on this PR?

Oh I see, ciflow/inductor

**Comment by shunting314** [comment]
I'm debugging the failure for CTCLoss. So far I found on CPU, the gradient for input is None while on GPU it's not. That cause the failure.

Right now CTCLoss is an instance of a parametrized test. I'll try to have an repro without parametrization. That may make the repro and fix easier.

**Pull request closed without merging**